src/Logary.Tests/Metrics.fs

```diff
@@ -0,0 +1,168 @@
+module Logary.Tests.Metrics
+open System
+open Expecto
+open Logary
+open Logary.Metric
+open Expecto.Flip
+
+let private basicConf = { name =  "items.in.queue"; description = "items in queue"; labelNames = [||] }
+let private gaugeConf = GaugeConf.create basicConf
+
+let tests = [
+
+  testCase "register metric" <| fun () ->
+    let registry = new MetricRegistry()
+    let gauge = gaugeConf |> registry.registerMetric
+    gauge.noLabels.inc 5.
+    let basicInfo, MetricInfo.Gauge gaugeInfo = gauge.noLabels.explore ()
+    basicInfo |> Expect.equal "should have same basic info" { name = gaugeConf.basicInfo.name; description = gaugeConf.basicInfo.description }
+    gaugeInfo.labels |> Expect.equal "should have empty labels" Map.empty
+    gaugeInfo.gaugeValue  |> Expect.equal  "should have same gauge value" 5.
+
+  testCase "register metric multi times" <| fun () ->
+    let registry = new MetricRegistry()
+    let gauge = gaugeConf |> registry.registerMetric
+    gauge.noLabels.inc 15.
+    let gauge = gaugeConf |> registry.registerMetric
+    gauge.noLabels.dec 10.
+
+    let _, MetricInfo.Gauge gaugeInfo = gauge.noLabels.explore ()
+    gaugeInfo.gaugeValue |> Expect.equal "should reuse the same one" 5.
+
+    Expect.throws "should throws when register metric with same name but different basic conf " <| fun () ->
+      GaugeConf.create {basicConf with labelNames = [| "with some label" |]} |> registry.registerMetric |> ignore
+
+  testCase "register metric with no labels" <| fun () ->
+    let registry = new MetricRegistry()
+    let gauge = gaugeConf |> registry.registerMetricWithNoLables
+    gauge.inc 10.
+    let gauge = gaugeConf |> registry.registerMetric
+    gauge.noLabels.inc 10.
+    let _, MetricInfo.Gauge gaugeInfo = gauge.noLabels.explore ()
+    gaugeInfo.gaugeValue |> Expect.equal "registerMetricWithNoLables is just a shotcut for registerMetric(gaugeConf).noLabels" 20.
+
+
+  testCase "register metric with labels" <| fun () ->
+
+    Expect.throws "should throws when gauge with label value but no label names set up" <| fun () ->
+        let registry = new MetricRegistry()
```

Comment:
2 space indent here

Reply:
fixed, BTW, do you know is there any auto formatting tools for fsharp ?